File: 0970-powerful-integers/0970-powerful-integers.py
class Solution(object):
    def powerfulIntegers(self, x, y, bound):
        xs = {x**i for i in range(20) if x**i < bound}
        ys = {y**j for j in range(20) if y**j < bound}
        return {x+y for x in xs for y in ys if x+y <= bound}
        

File: 0970-powerful-integers/0970-powerful-integers.cs
public class Solution {
    public IList<int> PowerfulIntegers(int x, int y, int bound) {
        HashSet<int> hs = new HashSet<int>();
        for (int i = 1; i < bound; i *= x)
        {
            for (int j = 1; i + j <= bound; j *= y)
            {
                hs.Add(i + j);
                if (y == 1) break;
            }
            if (x == 1) break;
        }
        return new List<int>(hs);
    }
}